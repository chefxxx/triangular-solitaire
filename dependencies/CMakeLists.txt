# Add all subdirectories recursively
file(GLOB children RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/*)

foreach(subdir ${children})
    if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${subdir})
        if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${subdir}/CMakeLists.txt)
            add_subdirectory(${subdir})
        endif()
    endif()
endforeach()