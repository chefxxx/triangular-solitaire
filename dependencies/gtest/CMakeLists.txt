# -------------------------------
# Fetch GTEST from github...
# -------------------------------

include(FetchContent)

FetchContent_Declare(
        libGTEST
        GIT_REPOSITORY https://github.com/google/googletest
        GIT_TAG main
        GIT_PROGRESS   TRUE
)

FetchContent_MakeAvailable(libGTEST)