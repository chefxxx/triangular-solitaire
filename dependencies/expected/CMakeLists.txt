#
#   Fetch "expected" from github...
#
message(STATUS "Adding expected...")
include(FetchContent)

FetchContent_Declare(
        expected
        GIT_REPOSITORY https://github.com/TartanLlama/expected.git
        GIT_TAG v1.1.0
)
FetchContent_MakeAvailable(expected)

# ----------------------------
# Interface Library
# ----------------------------
add_library(libExpected INTERFACE)
target_include_directories(libExpected INTERFACE ${expected_SOURCE_DIR}/include)